pkg/sql/logictest/logic.go

```diff
@@ -2105,98 +2130,106 @@ func (t *logicTest) execQuery(query logicQuery) error {
 	}
 	defer rows.Close()
 
-	cols, err := rows.Columns()
-	if err != nil {
-		return err
-	}
-	if len(query.colTypes) != len(cols) {
-		return fmt.Errorf("%s: expected %d columns, but found %d",
-			query.pos, len(query.colTypes), len(cols))
-	}
-	vals := make([]interface{}, len(cols))
-	for i := range vals {
-		vals[i] = new(interface{})
-	}
-
 	var actualResultsRaw []string
-	if query.colNames {
-		actualResultsRaw = append(actualResultsRaw, cols...)
-	}
-	for rows.Next() {
-		if err := rows.Scan(vals...); err != nil {
+	colLen := 0
```

Comment:
What we did for the `kvtrace` logic tests was to just put the column types etc in the query, like

`query T kvtrace`

Reply:
kvtrace is slightly different to this because it executes a sql query that gives the results as a single column, from what i could tell.

Reply:
Isn't that the same thing though, with actualResultsRaw = t.noticeBuffer?

```
query T noticetrace
....
----
NOTICE: aaaa
```